static/js/services/http-wrapper.js

```diff
@@ -32,13 +37,13 @@ var _ = require('underscore');
     '$http', '$q', 'ActiveRequests',
     function($http, $q, ActiveRequests) {
 
-      var wrap = function(args, fn) {
+      var wrap = function(args, fn, updateTarget) {
```

Comment:
I'm not 100% sure that state name helps - we have access to the previous and next state names in the `stateChangeStart` listener, but this won't help in differentiating between `left` and `right` update types.

I'm not 100% happy with the current `updateTarget` values - I'm sure there are more semantic names we could pick than `left` and `right`!